src/syft/core/store/store_interface.py

```diff
@@ -127,7 +116,17 @@ def __setitem__(self, key: UID, value: StorableObject) -> None:
 
     @syft_decorator(typechecking=True)
     def __delitem__(self, key: UID) -> None:
+        self.delete(key=key)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def delete(self, key: UID) -> None:
         """
+        We should write custom deletion code so we can check if the item exists
+        and then ensure deletion is called at a single place and that full __delitem__
+        is used where possible instead of the weaker del ref count.
```

Comment:
I think we should use ```del```.
Using ```__delitem__``` seems more like a hack we are doing because we can not find the problem :(